### PR TITLE
Use Prctl() and ioctl wrapper functions from x/sys/unix

### PR DESCRIPTION
**Pull request by tklauser**
Update golang.org/x/sys/unix once again to get some more newly added wrapper functions and switch libcontainer to use them instead of manually reimplemented local wrappers.

### CODE REVIEW

**Comment by tklauser** [comment]
Rebased on latest master to hopefully fix the shfmt issues which made Travis CI fail.

File: libcontainer/console_linux.go [hunk 1]
```diff
@@ -123,8 +123,8 @@ func unlockpt(f *os.File) error {
 
 // ptsname retrieves the name of the first available pts for the given master.
 func ptsname(f *os.File) (string, error) {
-	var n int32
-	if err := ioctl(f.Fd(), unix.TIOCGPTN, uintptr(unsafe.Pointer(&n))); err != nil {
+	n, err := unix.IoctlGetInt(int(f.Fd()), unix.TIOCGPTN)
```
Comment (dqminh) [inline libcontainer/console_linux.go]: Thanks ! Do you also want to make the same changes to https://github.com/containerd/console/blob/master/tc_linux.go ?

**Comment by cyphar** [comment]
LGTM. I know we've said this before, but :+1: for doing this work. :smile_cat:

File: libcontainer/console_linux.go (hunk 1 above)
Comment (tklauser) [inline libcontainer/console_linux.go]: Sure, will look into it and prepare a PR.

**Comment by cyphar** [comment]
The `shfmt` issue is not related to this PR. This is happening because we pull `shfmt` from master and they just changed some of their requirements. I'll send a PR to fix it.

**Comment by cyphar** [comment]
It looks like this is actually a bug in `shfmt`...

```
declare -F $completions_func >/dev/null && $completions_func
```

**Comment by tklauser** [comment]
Rebased on latest master with shfmt removed. Travis is now happy again :sunglasses:

**Comment by dqminh** [comment]
LGTM

**Comment by crosbymichael** [comment]
LGTM

**Pull request merged**